core/src/main/java/org/mule/runtime/core/message/DefaultEventBuilder.java

```diff
@@ -225,31 +216,21 @@ public Builder disableNotifications() {
   }
 
   @Override
-  @Deprecated
-  public Event.Builder refreshSync() {
-    this.synchronous = resolveEventSynchronicity();
-    this.modified = true;
-    return this;
-  }
-
-  @Override
   public Event build() {
     if (originalEvent != null && !modified) {
       return originalEvent;
     } else {
-      return new EventImplementation(context, message, flowVariables, exchangePattern, flow, session, transacted,
-                                     synchronous == null ? (resolveEventSynchronicity() && replyToHandler == null)
-                                         : synchronous,
+      return new EventImplementation(context, message, flowVariables, exchangePattern, flow, session,
+                                     synchronous == null ? resolveEventSynchronicity() : synchronous,
```

Comment:
This wasn't in 3.x, I checked,  seems it found its way in a refactor.  Current behaviour in 4.x relies on the fact that refreshSync does not take I to account reply to handler.